src/openrct2/ride/Vehicle.cpp

```diff
@@ -2363,60 +2363,20 @@ static void vehicle_update_waiting_for_passengers(rct_vehicle* vehicle)
                 return;
             }
 
+            // any load: load=4 , full: load=3,  3/4s: load=2, half: load=1, quarter: load=0
             uint8_t load = ride->depart_flags & RIDE_DEPART_WAIT_FOR_LOAD_MASK;
-            if (load == 3)
-            {
-                train_ready_to_depart(vehicle, num_peeps_on_train, num_used_seats_on_train);
-                return;
-            }
-
-            uint8_t three_quater_seats = (3 * num_seats_on_train) / 4;
-            if (three_quater_seats != 0 && num_peeps_on_train >= three_quater_seats)
-            {
-                vehicle->update_flags |= VEHICLE_UPDATE_FLAG_TRAIN_READY_DEPART;
-                train_ready_to_depart(vehicle, num_peeps_on_train, num_used_seats_on_train);
-                return;
-            }
-
-            if (load == 2)
-            {
-                train_ready_to_depart(vehicle, num_peeps_on_train, num_used_seats_on_train);
-                return;
-            }
-
-            if (num_seats_on_train / 2 != 0 && num_peeps_on_train >= num_seats_on_train / 2)
-            {
-                vehicle->update_flags |= VEHICLE_UPDATE_FLAG_TRAIN_READY_DEPART;
-                train_ready_to_depart(vehicle, num_peeps_on_train, num_used_seats_on_train);
-                return;
-            }
+            uint8_t peepTarget = ((load + 1) * num_seats_on_train + 3) / 4;
```

Comment:
`+3` looks very messy, and isn't in the original code.

Reply:
The intent was to compute ceiling( (load+1)/4 * num_seats), and the +3 is a trick to bypass the need of floats/doubles.

It's not in the original code because it did not actually wait for a quarter load. In #9970 there's a screenshot of vanilla RCT1 starting an 18-car train with 4 occupants at "wait for quarter load".

I mean, it's a bit of an argument over semantics, but I expected that 18-car train to leave at 6 occupants rather than 4. I could be convinced otherwise (removing the +3 reverts the behaviour to vanilla and doesn't affect the rest of the logic).

Reply:
I wrote out the results for a train with room for 18 people:

Load | "Real" result | RCT1/2 | This PR
--- | --- | --- | ---
Quarter | 4.5 | 4 | 5
Half | 9 | 9 | 9
Three q. | 13.5 | 13 | 14

Since the option is _minimum_ load, rounding up is indeed more in line with what people expect. I am assuming that the compiler will first do the multiplication and only then the addition.

I'd like to see the following things:
- A comment that explains why the +3 is there
- A changelog entry: `- Improved: [#9987] Minimum load rounding.`
- A network version bump

Reply:
Done, force pushed.